ipatests/test_webui/crypto_utils.py

```diff
@@ -2,26 +2,61 @@
 # Copyright (C) 2018  FreeIPA Contributors see COPYING for license
 #
 
+from datetime import datetime, timedelta
+
 from cryptography import x509
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives import serialization, hashes
 from cryptography.hazmat.primitives.asymmetric import rsa
 from cryptography.x509.oid import NameOID
 
 
-def generate_csr(hostname):
+def generate_csr(cn, is_hostname=True):
     key = rsa.generate_private_key(
         public_exponent=65537,
         key_size=2048,
         backend=default_backend()
     )
-    hostname = u'{}'.format(hostname)
+    cn = u'{}'.format(cn)
```

Comment:
It was actual for python2 because next function requires 'unicode' variable (hostname has 'str' type). Don't sure if there is needed to change it now, because if it will be backported to 4.6 we broke some tests.

Reply:
It will be more obvious if you use six.text_type.
six module is used throughout freeipa project.

Reply:
Another option is to add at the top of the module (there are many examples in other places)
```
if six.PY3:
    unicode = str

Reply:
What is bad in current version: function accepts string and unicode in PY2 and bytes and str in PY3. But for  bytes in PY it produces invalid result:
```
In [1]: b = b'abcd'

In [2]: str(b)
Out[2]: "b'abcd'"

In [3]: u'{}'.format(b)
Out[3]: "b'abcd'"
```

Reply:
To make function accept both binary and unicode strings you could write something like:
```
if isinstance(cn, six.text_type):
    cn = cn.decode()
```

or

```
if hasattr(cn, 'decode'):
    cn = cn.decode()
```